integration-tests/external-plugin-otel-test/external-plugin-otel-test-app/src/main/java/testapp/Main.java

```diff
@@ -48,11 +50,27 @@ private static void span() {
     }
 
     private static void checkCurrentSpanVisibleThroughOTel() {
-        SpanContext spanContext = Span.current().getSpanContext();
-        if (!spanContext.isValid()) {
-            System.out.println("no active OTel Span");
-        } else {
-            System.out.printf("active span ID = %s, trace ID = %s%n", spanContext.getSpanId(), spanContext.getTraceId());
+        if (OTEL_ON_CLASSPATH) {
+            new Runnable() {
+                @Override
+                public void run() {
+                    SpanContext spanContext = Span.current().getSpanContext();
+                    if (!spanContext.isValid()) {
+                        System.out.println("no active OTel Span");
+                    } else {
+                        System.out.printf("active span ID = %s, trace ID = %s%n", spanContext.getSpanId(), spanContext.getTraceId());
+                    }
+                }
+            }.run();
```

Comment:
Does wrapping into an anonymous class allows to not load it directly from the parent class ? If so then maybe it would be worth adding a small comment for the futures ourselves that might wonder why.

Reply:
It is not possible to load a class which contains references to classes which are not on the classpath.
So if we would run this class without the `Runnable`-wrapping and without Otel on the classpath, the class would fail to load. This is prevent by moving the Otel-calls to another class (anonymous in this case). This anonymous class is never loaded when we run without Otel because we first make sure that Otel is actually on the classpath. This way, it is possible to run the App without problems with and without Otel on the classpath.

I'll add a comment to explain this behaviour.